### PR TITLE
fix: change textColor to color, add pseudo styles for Text

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Breaking Change(Text): changed `textColor` prop to `color`
- [x] Breaking Change(Style-Props): changed `textColor` type to `color`
- [x] Setup pseudo styles for `Text`
- [x] Update tests and stories to use `color`
- [x] Update website to use `color`

### CODE REVIEW

File: packages/paste-website/src/pages/primitives/text/index.mdx [hunk 1]
```diff
@@ -234,10 +246,13 @@ All the regular HTML attributes (`role`, `aria-*`, `type`, and so on) including
...
-| textColor?      | `ResponsiveValue<keyof ThemeShape['textColors']>`                             |                           |         |
+| color?          | `ResponsiveValue<keyof ThemeShape['textColors']>`                             |                           |         |
 | textDecoration? | `ResponsiveValue<CSS.TextDecorationProperty<CSS.TextDecorationLineProperty>>` |                           |         |
 | textOverflow?   | `ResponsiveValue<CSS.TextOverflowProperty>`                                   |                           |         |
 | whitespce?      | `ResponsiveValue<CSS.WhiteSpaceProperty>`                                     |                           |         |
+| _hover?         | TextProps                                                                     |                           |         |
+| _active?        | TextProps                                                                     |                           |         |
+| _focus?         | TextProps                                                                     |                           |         |
```
Comment (SiTaggart) [inline packages/paste-website/src/pages/primitives/text/index.mdx:255]: Just need to add the other ones we added here

File: packages/paste-core/primitives/text/stories/index.stories.tsx [hunk 2]
```diff
@@ -102,9 +102,22 @@ storiesOf('Primitives|Text', module)
...
         padding={['space0', 'space10', 'space130', 'space70']}
         textAlign={['center', 'right', 'left']}
-        textColor={['colorTextBrandHighlight', 'colorTextLink', 'colorTextSuccess', 'colorText']}
+        color={['colorTextBrandHighlight', 'colorTextLink', 'colorTextSuccess', 'colorText']}
       >
         Some text with responsive styling.
       </Text>
     );
+  })
+  .add('Pseudo-classes with props', () => {
```
Comment (SiTaggart) [inline packages/paste-core/primitives/text/stories/index.stories.tsx:111]: I wouldn't mind an additional test or 2 in addition to the story. Just to prove that the pseudo selectors are working and outputting css we expect.

Comment (richbachman) [reply packages/paste-core/primitives/text/stories/index.stories.tsx]: I've added a few more stories. Let me know if those will work.

Comment (SiTaggart) [reply packages/paste-core/primitives/text/stories/index.stories.tsx]: Oh, I meant some jest tests, maybe snapshots. Something that proves the code produces the css you expect

Comment (richbachman) [reply packages/paste-core/primitives/text/stories/index.stories.tsx]: Oh I see that now. I read that wrong. I removed those stories and added some snapshot tests.

Comment (richbachman) [reply packages/paste-core/primitives/text/stories/index.stories.tsx]: Updated tests to test all pseudo-classes.

File: packages/paste-core/primitives/text/src/PseudoPropStyles.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export const PseudoPropStyles: {[key: string]: string} = {
+  _hover: '&:hover',
+  _active: '&:active, &[data-active=true]',
+  _focus: '&:focus',
+  _visited: '&:visited',
+  _even: '&:nth-of-type(even)',
+  _odd: '&:nth-of-type(odd)',
+  _disabled:
+    '&:disabled, &:disabled:focus, &:disabled:hover, &[aria-disabled=true], &[aria-disabled=true]:focus, &[aria-disabled=true]:hover',
+  _checked: '&[aria-checked=true]',
```
Comment (SiTaggart) [inline packages/paste-core/primitives/text/src/PseudoPropStyles.ts:10]: This should probably at least cover `&:checked` as well

File: packages/paste-core/primitives/text/src/PseudoPropStyles.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  _mixed: '&[aria-checked=mixed]',
```
Comment (SiTaggart) [inline packages/paste-core/primitives/text/src/PseudoPropStyles.ts:11]: This should probably cover `&:indeterminate ` as well

File: packages/paste-core/primitives/text/src/PseudoPropStyles.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  _selected: '&[aria-selected=true]',
+  _invalid: '&[aria-invalid=true]',
```
Comment (SiTaggart) [inline packages/paste-core/primitives/text/src/PseudoPropStyles.ts:13]: `&:invalid`

**Comment by richbachman** [comment]
I've added the missing pseudo selectors to `Text` and `Box`.

**Pull request merged**